internal/common/context.go

```diff
@@ -0,0 +1,51 @@
+package common
+
+import (
+	"context"
+
+	"github.com/sirupsen/logrus"
+)
+
+type ctxKey int
+
+const (
+	requestIdCtx         ctxKey = iota
+	insightsRequestIdCtx ctxKey = iota
+	requestDataCtx       ctxKey = iota
+)
+
+func WithRequestId(ctx context.Context, requestId string) context.Context {
+	return context.WithValue(ctx, requestIdCtx, requestId)
```

Comment:
Opportunity for future: https://github.com/RedHatInsights/platform-go-middlewares/blob/44143f6300d86c94a78cdfc5ad5bbaf66b60d53d/request_id/request_id.go#L69C6-L69C25 ? :)

Reply:
This must come with v2 upgrade first. I can probably do it.